homeassistant/components/sensor/gtfs.py

```diff
@@ -0,0 +1,249 @@
+    AND start_station.stop_id = '{}' AND end_station.stop_id = '{}'
+    ORDER BY origin_stop_time.departure_time LIMIT 1;"""\
+    .format(day_name, now_str, origin_station.id, destination_station.id)
+    result = sched.engine.execute(sql_query)
+    item = {}
+    for row in result:
+        item = row
+
+    today = datetime.datetime.today().strftime("%Y-%m-%d")
+    departure_time_string = "{} {}".format(today, item[2])
+    arrival_time_string = "{} {}".format(today, item[3])
+    departure_time = datetime.datetime.strptime(departure_time_string,
+                                                TIME_FORMAT)
+    arrival_time = datetime.datetime.strptime(arrival_time_string,
+                                              TIME_FORMAT)
+
+    seconds_until = (departure_time-datetime.datetime.now()).total_seconds()
+    minutes_until = int(seconds_until / 60)
+
+    route = sched.routes_by_id(item[1])[0]
+
+    origin_stoptime_arrival_time = "{} {}".format(today, item[4])
+
+    origin_stoptime_departure_time = "{} {}".format(today, item[5])
+
+    dest_stoptime_arrival_time = "{} {}".format(today, item[11])
+
+    dest_stoptime_depart_time = "{} {}".format(today, item[12])
+
+    origin_stop_time_dict = {
+        "Arrival Time": origin_stoptime_arrival_time,
+        "Departure Time": origin_stoptime_departure_time,
+        "Drop Off Type": item[6], "Pickup Type": item[7],
+        "Shape Dist Traveled": item[8], "Headsign": item[9],
+        "Sequence": item[10]
+    }
+
+    destination_stop_time_dict = {
+        "Arrival Time": dest_stoptime_arrival_time,
+        "Departure Time": dest_stoptime_depart_time,
+        "Drop Off Type": item[13], "Pickup Type": item[14],
+        "Shape Dist Traveled": item[15], "Headsign": item[16],
+        "Sequence": item[17]
+    }
+
+    return {
+        "trip_id": item[0],
+        "trip": sched.trips_by_id(item[0])[0],
+        "route": route,
+        "agency": sched.agencies_by_id(route.agency_id)[0],
+        "origin_station": origin_station,
+        "departure_time": departure_time,
+        "destination_station": destination_station,
+        "arrival_time": arrival_time,
+        "seconds_until_departure": seconds_until,
+        "minutes_until_departure": minutes_until,
+        "origin_stop_time": origin_stop_time_dict,
+        "destination_stop_time": destination_stop_time_dict
+    }
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Get the GTFS sensor."""
+    if config.get("origin") is None:
+        _LOGGER.error("Origin must be set in the GTFS configuration!")
+        return False
+
+    if config.get("destination") is None:
+        _LOGGER.error("Destination must be set in the GTFS configuration!")
+        return False
+
+    if config.get("data") is None:
+        _LOGGER.error("Data must be set in the GTFS configuration!")
+        return False
+
+    dev = []
+    dev.append(GTFSDepartureSensor(config["data"], hass.config.path("gtfs"),
```

Comment:
No, and actually I just realized there isn't even checking to see if the path exists or is valid. :( Will add validation and the folder creation shortly.